pathfinder.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"runtime"
+	"sync"
+
+	"golang.org/x/mod/module"
+
+	"github.com/CrowdStrike/perseus/perseusapi"
+)
+
+// newPathFinder initializes and returns a new [pathFinder] instance using the provided Perseus
+// client, maximum depth, and status callback.
+func newPathFinder(c perseusapi.PerseusServiceClient, maxDepth int, status func(string)) pathFinder {
+	return pathFinder{
+		c:        c,
```

Comment:
You could drop the field names in this case if you're into that.

Reply:
I like to always used named fields in initializers.